srp_daemon/srp_daemon.c

```diff
@@ -2036,6 +2036,18 @@ int main(int argc, char *argv[])
 	int			lockfd = -1;
 	int			received_signal = 0;
 
+#ifndef __CHECKER__
+	/*
+	 * Hide these checks for sparse because sparse gets most of them wrong.
+	 */
+	STATIC_ASSERT(sizeof(ib_sa_mad_t) == 256);
+	STATIC_ASSERT(sizeof(ib_path_rec_t) == 64);
+	STATIC_ASSERT(sizeof(ib_mad_t) == 24);
+	STATIC_ASSERT(sizeof(ib_inform_info_t) == 36);
+	STATIC_ASSERT(sizeof(ib_mad_notice_attr_t) == 80);
+	STATIC_ASSERT(offsetof(ib_mad_notice_attr_t,
+			       data_details.ntc_257_258.gid2) == 45);
```

Comment:
I missed where it's used. Would you point me at it ? I see the other notices in srp_handle_traps.c but not this one.

If it needs to be preserved, it's wrong/not according to IBA spec. It was lifted from old ib_types.h and needs fixing. It's why offset of gid there is mistakenly 45!